readthedocs/search/api.py

```diff
@@ -79,9 +80,11 @@ def get_queryset(self):
         kwargs['filters']['project'] = [p.slug for p in self.get_all_projects()]
         kwargs['filters']['version'] = self.request.query_params.get('version')
         if not kwargs['filters']['project']:
-            raise ValidationError("Unable to find a project to search")
+            log.info("Unable to find a project to search")
+            return HTMLFile.objects.none()
         if not kwargs['filters']['version']:
-            raise ValidationError("Unable to find a version to search")
+            log.info("Unable to find a version to search")
+            return HTMLFile.objects.none()
```

Comment:
Some lines below, we are returning a `PageSearch` object. Shouldn't we return something similar here instead of a different queryset object?

Reply:
`PageSearch` isn't an django model, is an elastic search document based on `HTMLFile`. Actually it's weird becuase the `PagaSearch` has an interface similar to what django rest framework expects (there is a comment for that in this class or somewhere else). Also, funny that returning an empty `PageSearch` doesn't work with django rest framework.

From what I saw, we can return real `HTMLFile` objects, but that requires a query to the db, so I guess that's why we are not doing that and doing this hacky thing.

Reply:
There's also an open issue related to this: #5235